nibabel/brikhead.py

```diff
@@ -0,0 +1,558 @@
+    array([[ -3.    ,  -0.    ,  -0.    ,  49.5   ],
+           [ -0.    ,  -3.    ,  -0.    ,  82.312 ],
+           [  0.    ,   0.    ,   3.    , -52.3511],
+           [  0.    ,   0.    ,   0.    ,   1.    ]])
+    >>> head = load(os.path.join(datadir, 'example4d+orig.HEAD'))
+    >>> np.array_equal(head.get_data(), brik.get_data())
+    True
+    """
+
+    header_class = AFNIHeader
+    valid_exts = ('.brik', '.head')
+    files_types = (('image', '.brik'), ('header', '.head'))
+    _compressed_suffixes = ('.gz', '.bz2')
+    makeable = False
+    rw = False
+    ImageArrayProxy = AFNIArrayProxy
+
+    @classmethod
+    @kw_only_meth(1)
+    def from_file_map(klass, file_map, mmap=True):
+        """
+        Creates an AFNIImage instance from `file_map`
+
+        Parameters
+        ----------
+        file_map : dict
+            dict with keys ``image, header`` and values being fileholder
+            objects for the respective BRIK and HEAD files
+        mmap : {True, False, 'c', 'r'}, optional, keyword only
+            `mmap` controls the use of numpy memory mapping for reading image
+            array data.  If False, do not try numpy ``memmap`` for data array.
+            If one of {'c', 'r'}, try numpy memmap with ``mode=mmap``.  A
+            `mmap` value of True gives the same behavior as ``mmap='c'``.  If
+            image data file cannot be memory-mapped, ignore `mmap` value and
+            read array from file.
+        """
+        with file_map['header'].get_prepare_fileobj('rt') as hdr_fobj:
+            hdr = klass.header_class.from_fileobj(hdr_fobj)
+        imgf = file_map['image'].fileobj
+        if imgf is None:
+            imgf = file_map['image'].filename
+        data = klass.ImageArrayProxy(imgf, hdr.copy(), mmap=mmap)
+        return klass(data, hdr.get_affine(), header=hdr, extra=None,
+                     file_map=file_map)
+
+    @classmethod
+    @kw_only_meth(1)
+    def from_filename(klass, filename, mmap=True):
+        """
+        Creates an AFNIImage instance from `filename`
+
+        Parameters
+        ----------
+        filename : str
+            Path to BRIK or HEAD file to be loaded
+        mmap : {True, False, 'c', 'r'}, optional, keyword only
+            `mmap` controls the use of numpy memory mapping for reading image
+            array data.  If False, do not try numpy ``memmap`` for data array.
+            If one of {'c', 'r'}, try numpy memmap with ``mode=mmap``.  A
+            `mmap` value of True gives the same behavior as ``mmap='c'``.  If
+            image data file cannot be memory-mapped, ignore `mmap` value and
+            read array from file.
+        """
+        file_map = klass.filespec_to_file_map(filename)
+        return klass.from_file_map(file_map, mmap=mmap)
+
+    @classmethod
+    def filespec_to_file_map(klass, filespec):
+        """
+        Make `file_map` from filename `filespec`
+
+        Deals with idiosyncracies of AFNI BRIK / HEAD formats
```

Comment:
Maybe add something like:

AFNI BRIK files can be compressed, but HEAD files cannot - see https://afni.nimh.nih.gov/pub/dist/doc/program_help/README.compression.html.  Let's say you have an AFNI file you want to load, with files `my_image.HEAD` and `my_image.BRIK.gz`. To identify the AFNI BRIK / HEAD pair, you can specify:

* The HEAD filename - e.g. ``my_image.HEAD``
* The BRIK filename without compressed extension, e.g. ``my_image.BRIK``
* The full BRIK filename - e.g. ``my_image.BRIK.gz``.

As a side note, I guess you're not proposing to support `.Z` compressed files.  I bet they are rare.